jmeter/src/main/java/com/excelian/mache/jmeter/cassandra/knownKeys/ReadFromCacheUntilItMatchesValue.java

```diff
@@ -0,0 +1,61 @@
+package com.excelian.mache.jmeter.cassandra.knownKeys;
+
+import com.excelian.mache.jmeter.cassandra.CassandraTestEntity;
+import com.excelian.mache.jmeter.cassandra.MacheAbstractCassandraKafkaSamplerClient;
+import org.apache.jmeter.config.Arguments;
+import org.apache.jmeter.protocol.java.sampler.JavaSamplerContext;
+import org.apache.jmeter.samplers.SampleResult;
+
+import java.util.Map;
+
+public class ReadFromCacheUntilItMatchesValue extends ReadFromCache {
+	private static final long serialVersionUID = -8612323545680365704L;
+
+	@Override
+	public void setupTest(JavaSamplerContext context) {
+		super.setupTest(context);
+	}
+
+	@Override
+	public void teardownTest(JavaSamplerContext context) {
+		super.teardownTest(context);
+	}
+
+	@Override
+	public SampleResult runTest(JavaSamplerContext context)  {
+		return super.runTest(context);
+	}
+
+	@Override
+	protected void readFromCache(final Map<String, String> params) throws Exception {
+		final String docNumber = params.get("entity.keyNo");
+		final String key = "document_" + docNumber;
+		final String expectedValue=params.get("entity.expectedValue");
+
+		Long time=System.currentTimeMillis();
+
+		do {
+			CassandraTestEntity result=cache1.get(key);
+			if(result.aString.equals(expectedValue))
+			{
+			//	getLogger().info("# Read k="+key+" v="+expectedValue);
+				break;
+			}
+
+			if((System.currentTimeMillis()-time)>2*60*1000)//We don't use a future as we want to stay in same thread.
```

Comment:
consider changing to : `int TWO_MINUTES = 2*60*1000;`

or `2 * TimeUnit.MINUTES.toMillis()`